fn/io.go

```diff
@@ -0,0 +1,45 @@
+package fn
+
+import (
+	"os"
+)
+
+// WriteFile synchronously writes data to the named file.
+// If the file does not exist, WriteFile creates it with permissions perm
+// (before umask); otherwise WriteFile truncates it before writing, without
+// changing permissions.
+// By opening the file with O_SYNC, it ensures the data is written to disk.
+// If an error occurs, it does not remove the file.
+func WriteFile(name string, data []byte, perm os.FileMode) error {
+	f, err := os.OpenFile(
+		name, os.O_WRONLY|os.O_CREATE|os.O_TRUNC|os.O_SYNC, perm,
+	)
+	if err != nil {
+		return err
+	}
+
+	_, err = f.Write(data)
+
+	// Prioritize the error on Write but make sure to call Close regardless
+	// to avoid leaking a file handle.
+	if err1 := f.Close(); err1 != nil && err == nil {
+		err = err1
+	}
+
+	return err
+}
+
+// WriteFile synchronously writes data to the named file.
```

Comment:
```suggestion
// WriteFileRemove synchronously writes data to the named file.
```

Reply:
Oh! I didn't realize that typo.

Fixed!